version: "3.4"
   
services:
  db:
   build: 
     context: db
     dockerfile: Dockerfile
   container_name: db
   volumes:
      - ./data/db:/var/lib/postgresql/data
   environment:
    - POSTGRES_USER=gisuser
    - DATABASE_PASS
    - POSTGRES_DBNAME=gis
   ports:
   - "5432:5432"
   healthcheck:
        test: ["CMD", "curl", "-f", "http://localhost:5432"]
        interval: 30s
        timeout: 10s
        retries: 5

  migrate:
    build: 
     context: backend
    container_name: migrate
    command: python manage.py migrate
    volumes:
      - ./backend/:/src/app
    environment:
      - DATABASE_HOST=db
      - POSTGRES_USER=gisuser
      - DATABASE_PASS
      - POSTGRES_DBNAME=gis
    restart: on-failure
    depends_on: 
      - db
    links: 
      - db
  

  web:
    build: 
     context: backend
     dockerfile: Dockerfile
    container_name: backend
    command: python manage.py runserver 0.0.0.0:8000
    environment:
      - DATABASE_HOST=db
      - POSTGRES_USER=gisuser
      - DATABASE_PASS
      - POSTGRES_DBNAME=gis
      - AWS_SECRET_ACCES_KEY
      - AWS_ACCES_KEY_ID
    volumes:
      - ./backend/:/src/app
    ports:
      - "8000:8000"
    depends_on:
      - migrate
    links:
      - db
    restart: on-failure
